### PR TITLE
OLS-186: Use `@patch` instead of monkeypatch

### PR DESCRIPTION
**Pull request by tisnik**
## Type of change

- [x] PoC
- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Description

PoC: use patch instead of monkeypatch

### CODE REVIEW

File: tests/unit/query_helpers/test_question_validator.py [hunk 1]
```diff
@@ -16,18 +18,16 @@ def question_validator():
...
     # [INVALID,NOYAML]
     # [VALID,NOYAML]
     # [VALID,YAML]
 
-    ml = mock_llm_chain({"text": "default"})
-    monkeypatch.setattr(ols.src.query_helpers.question_validator, "LLMChain", ml)
-    monkeypatch.setattr(
-        ols.src.query_helpers.question_validator, "LLMLoader", mock_llm_loader()
-    )
+    llm_chain = mock_llm_chain({"text": "default"})
```
Comment (bparees) [inline tests/unit/query_helpers/test_question_validator.py:30]: this is somehow replacing the llm_chain variable in question_validator.py?  i'm not understand how this works.

but it certainly looks cleaner

File: tests/unit/query_helpers/test_question_validator.py [hunk 2]
```diff
@@ -16,36 +17,34 @@ def question_validator():
...
             conversation="1234", query="What is the meaning of life?"
         )
 
 
-def test_valid_responses(question_validator, monkeypatch):
+@patch("ols.src.query_helpers.question_validator.LLMLoader", new=mock_llm_loader(None))
+def test_valid_responses(question_validator):
     """Test how valid responses are handled by QuestionValidator."""
     for retval in ["INVALID,NOYAML", "VALID,NOYAML", "VALID,YAML"]:
+        # basically `@patch` and `with patch():` do the same thing, but the later
```
Comment (bparees) [inline tests/unit/query_helpers/test_question_validator.py:42]:
```suggestion
        # basically `@patch` and `with patch():` do the same thing, but the latter
```

**Comment by bparees** [comment]
@tisnik did you intend for this to merge as is(if so please squash the commits), or was this just to show @thoraxe and I what the change would look like?

**Comment by tisnik** [comment]
> @tisnik did you intend for this to merge as is(if so please squash the commits), or was this just to show @thoraxe and I what the change would look like?

I will squash it + remove PoC from title, to merge it @bparees

**Pull request merged**